Classes/Updater.lua

```diff
@@ -0,0 +1,144 @@
+
+-- Updater.lua
+
+-- Contains the cUpdater class used to CheckForNewerVersion if there is a newer version of WorldEdit available
+
+
+
+
+cUpdater = {}
+
+
+
+
+local g_CheckAttempts = 0
+local g_PluginInfo = ""
+local g_LatestVersion = nil
+local g_DownloadFile = nil
+
+
+
+
+local VersionCheckForNewerVersionerCallbacks =
+{
+	OnConnected = function(a_TCPLink)
+		local res, msg = a_TCPLink:StartTLSClient("", "", "")
+		a_TCPLink:Send("GET /cuberite/WorldEdit/master/Info.lua HTTP/1.1\r\nHost: raw.githubusercontent.com\r\nConnection: Close\r\n\r\n")
+	end,
+	
+	OnError = function(a_TCPLink, a_ErrorCode, a_ErrorMsg) 
+		LOGWARNING("Error while checking for newer WorldEdit version: " .. a_ErrorMsg)
+	end,
+	
+	OnReceivedData = function(a_TCPLink, a_Data)
+		g_PluginInfo = g_PluginInfo .. a_Data
+	end,
+	
+	OnRemoteClosed = function(a_TCPLink)
+		cUpdater:ParseResults()
+	end,
+}
+
+
+
+
+
+local NewVersionDownloaderCallbacks =
+{
+	OnConnected = function(a_TCPLink)
+		local res, msg = a_TCPLink:StartTLSClient("", "", "")
+		a_TCPLink:Send("GET /cuberite/WorldEdit/zip/master HTTP/1.1\r\nHost: codeload.github.com\r\nConnection: Close\r\n\r\n")
+		g_DownloadFile = io.open("Plugins/WorldEdit v" .. (g_LatestVersion or "_Unknown") .. ".zip", "wb")
+	end,
+	
+	OnError = function(a_TCPLink, a_ErrorCode, a_ErrorMsg)
+		LOGWARNING("Error while downloading newer worldedit version: " .. a_ErrorCode)
+		g_DownloadFile:close()
+	end,
+	
+	OnReceivedData = function(a_TCPLink, a_Data)
+		g_DownloadFile:write(a_Data:match("^.-\r\n\r\n(.*)") or a_Data)
+	end,
+	
+	OnRemoteClosed = function(a_TCPLink)
+		g_DownloadFile:close()
+	end,	
+}
+
+
+
+
+
+function cUpdater:CheckForNewerVersion()
+	if (g_CheckAttempts < g_Config.Updates.NumAttempts) then
+		g_CheckAttempts = g_CheckAttempts + 1
+		cNetwork:Connect("raw.githubusercontent.com", 443, VersionCheckForNewerVersionerCallbacks)
+	else
+		LOGWARNING("Could not connect to Github to check for a newer version for WorldEdit")
```

Comment:
That's one tangled code-flow, I had to re-read this thing three times before I found where the actual looping is done.

I think it would be better to move the count check to the `ParseResults()` function, it would make more sense to me - "if downloading failed, retry, unless too many retries"